### PR TITLE
Add couch_util:validate_callback_exists

### PR DESCRIPTION
**Pull request by iilyak**
COUCHDB-2585

### CODE REVIEW

File: src/couch_util.erl [hunk 1]
```diff
@@ -537,3 +538,34 @@ match_rest_of_prefix([{Pos, _Len} | Rest], Prefix, Data, PrefixLength, N) ->
         {_Pos, _Len1} ->
             {partial, N + Pos}
     end.
+
+callback_exists(Module, Function, Arity) ->
+    case ensure_loaded(Module) of
+    true ->
+        InfoList = Module:module_info(exports),
+        lists:any(fun(E) -> E == {Function, Arity} end, InfoList);
```
Comment (kxepal) [inline src/couch_util.erl]: It looks like you really want to use `lists:member/2` here (;

**Comment by kxepal** [comment]
Nice helper!

**Comment by janl** [comment]
how is this related to https://issues.apache.org/jira/browse/COUCHDB-2585?

**Comment by janl** [comment]
Ah, found it, https://issues.apache.org/jira/browse/COUCHDB-2625 was probably meant as a reference.

**Comment by kxepal** [comment]
@iilyak could you squash your commits please? I'll merge it.

**Comment by iilyak** [comment]
Squashed

**Pull request merged**